mlflow/java/src/main/java/org/mlflow/sagemaker/ScoringServer.java

```diff
@@ -0,0 +1,291 @@
+      // The server connector port request returned an error code
+      return Optional.empty();
+    }
+  }
+
+  public static class ServerStateChangeException extends RuntimeException {
+    ServerStateChangeException(Exception e) {
+      super(e.getMessage());
+    }
+  }
+
+  /**
+   * Configuration providing the minimum and maximum number of threads to allocate to the scoring
+   * server
+   */
+  static class ServerThreadConfiguration {
+    static final String ENV_VAR_MINIMUM_SERVER_THREADS = "SCORING_SERVER_MIN_THREADS";
+    static final String ENV_VAR_MAXIMUM_SERVER_THREADS = "SCORING_SERVER_MAX_THREADS";
+
+    static final int DEFAULT_MINIMUM_SERVER_THREADS = 1;
+    // Assuming an 8 core machine with hyperthreading
+    static final int DEFAULT_MAXIMUM_SERVER_THREADS = 16;
+
+    private final int minThreads;
+    private final int maxThreads;
+
+    private ServerThreadConfiguration(int minThreads, int maxThreads) {
+      this.minThreads = minThreads;
+      this.maxThreads = maxThreads;
+    }
+
+    static ServerThreadConfiguration create(Environment environment) {
+      int minThreads =
+          environment.getIntegerValue(
+              ENV_VAR_MINIMUM_SERVER_THREADS, DEFAULT_MINIMUM_SERVER_THREADS);
+      int maxThreads =
+          environment.getIntegerValue(
+              ENV_VAR_MAXIMUM_SERVER_THREADS, DEFAULT_MAXIMUM_SERVER_THREADS);
+      return new ServerThreadConfiguration(minThreads, maxThreads);
+    }
+
+    /** @return The minimum number of server threads specified by the configuration */
+    int getMinThreads() {
+      return this.minThreads;
+    }
+
+    /** @return The maximum number of server threads specified by the configuration */
+    int getMaxThreads() {
+      return this.maxThreads;
+    }
+  }
+
+  static class PingServlet extends HttpServlet {
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) {
+      response.setStatus(HttpServletResponse.SC_OK);
+    }
+  }
+
+  static class InvocationsServlet extends HttpServlet {
+    private final Predictor predictor;
+
+    InvocationsServlet(Predictor predictor) {
+      this.predictor = predictor;
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response)
+        throws IOException {
+      String requestContentType = request.getHeader("Content-type");
+      String requestBody = FileUtils.readInputStreamAsUtf8(request.getInputStream());
+      String responseContent = null;
+      try {
+        responseContent = evaluateRequest(requestBody, requestContentType);
+      } catch (PredictorEvaluationException e) {
+        logger.error("Encountered a failure when evaluating the predictor.", e);
+        response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+        responseContent = getErrorResponseJson(e.getMessage());
+      } catch (InvalidRequestTypeException e) {
+        response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
```

Comment:
This one is also kind of weird, maybe we can log an info or something.

Reply:
Added a `logger.info` statement